flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/UnionResultSubpartitionViewTest.java

```diff
@@ -0,0 +1,182 @@
+        view1 = new TestingResultSubpartitionView(view, buffers1);
+        view.notifyViewCreated(1, view1);
+    }
+
+    @Test
+    void testGetNextBuffer() throws IOException {
+        assertThat(view.getNextBuffer()).isNull();
+        view0.notifyDataAvailable();
+        ResultSubpartition.BufferAndBacklog bufferAndBacklog = view.getNextBuffer();
+        assertThat(bufferAndBacklog.buffer()).isEqualTo(buffers0.get(0));
+        assertThat(bufferAndBacklog.buffersInBacklog()).isEqualTo(buffers0.size() - 1);
+
+        view1.notifyDataAvailable();
+        assertThat(view.getNextBuffer().buffer()).isEqualTo(buffers0.get(1));
+
+        List<Buffer> buffers = new ArrayList<>();
+        while (view.getAvailabilityAndBacklog(true).isAvailable()) {
+            buffers.add(view.getNextBuffer().buffer());
+        }
+        assertThat(buffers)
+                .hasSize(buffers0.size() + buffers1.size() - 2)
+                .containsSubsequence(buffers0.subList(2, buffers0.size()))
+                .containsSubsequence(buffers1);
+    }
+
+    @Test
+    void testGetAvailabilityAndBacklog() throws IOException {
+        view0.notifyDataAvailable();
+        view1.notifyDataAvailable();
+
+        ResultSubpartitionView.AvailabilityWithBacklog availabilityAndBacklog1 =
+                view.getAvailabilityAndBacklog(false);
+        assertThat(availabilityAndBacklog1.getBacklog()).isPositive();
+        assertThat(availabilityAndBacklog1.isAvailable()).isEqualTo(false);
+        ResultSubpartitionView.AvailabilityWithBacklog availabilityAndBacklog2 =
+                view.getAvailabilityAndBacklog(true);
+        assertThat(availabilityAndBacklog2.getBacklog()).isPositive();
+        assertThat(availabilityAndBacklog2.isAvailable()).isEqualTo(true);
+
+        for (int i = 1; i < buffers0.size() + buffers1.size(); i++) {
+            view.getNextBuffer();
+        }
+
+        ResultSubpartitionView.AvailabilityWithBacklog availabilityAndBacklog3 =
+                view.getAvailabilityAndBacklog(false);
+        assertThat(availabilityAndBacklog3.getBacklog()).isEqualTo(0);
+        assertThat(availabilityAndBacklog3.isAvailable()).isEqualTo(true);
+        ResultSubpartitionView.AvailabilityWithBacklog availabilityAndBacklog4 =
+                view.getAvailabilityAndBacklog(true);
+        assertThat(availabilityAndBacklog4.getBacklog()).isEqualTo(0);
+        assertThat(availabilityAndBacklog4.isAvailable()).isEqualTo(true);
+    }
+
+    @Test
+    void testReleaseAllResources() throws IOException {
+        assertThat(view.isReleased()).isFalse();
+        assertThat(view0.isReleased()).isFalse();
+        assertThat(view1.isReleased()).isFalse();
+
+        view.releaseAllResources();
+
+        assertThat(view.isReleased()).isTrue();
+        assertThat(view0.isReleased()).isTrue();
+        assertThat(view1.isReleased()).isTrue();
+    }
+
+    private static class TestingResultSubpartitionView extends NoOpResultSubpartitionView {
```

Comment:
We can define a new `TestingResultSubpartitionView` class(Ref `TestingHsDataView`) instead of extending from `NoOpResultSubpartitionView`, then we can reuse it when needed.

Reply:
A more common practice is to create a subclass of `NoOpResultSubpartitionView`, like `DefaultBufferResultSubpartitionView`, `EmptyAlwaysAvailableResultSubpartitionView`, and `NextIsEventResultSubpartitionView`. We are not sure what other tests need in a `ResultSubpartitionView`, so it might be hard to determine what the new `TestingResultSubpartitionView` class should look like.